sqlite.go

```diff
@@ -490,14 +490,24 @@ func (s *stmt) ExecContext(ctx context.Context, args []driver.NamedValue) (drive
 	if err := s.bindAll(args); err != nil {
 		return nil, s.reserr("Stmt.Exec(Bind)", err)
 	}
+	var done chan struct{}
 	if ctx.Value(queryCancelKey{}) != nil {
-		var cancel context.CancelFunc
-		ctx, cancel = context.WithCancel(ctx)
-		defer cancel()
+		done = make(chan struct{})
+		pctx, pcancel := context.WithCancel(ctx)
+		defer pcancel() // to make the AfterFunc fire and close(done)
 
 		db := s.stmt.DBHandle()
-		go func() { <-ctx.Done(); db.Interrupt() }()
+		stop := context.AfterFunc(pctx, func() {
+			defer close(done)
+			if ctx.Err() != nil {
+				db.Interrupt()
+			}
+		})
+		// In the event we get an error from the initial step and exit early,
```

Comment:
still gonna clarify "initial step" into something that sounds more sqlite-specific?

Reply:
Oops, yes, sorry I missed this one.